### PR TITLE
doc: add explanation about python support

### PR DESCRIPTION
**Pull request by aciba90**
Port doc/design.md to RST to natively render tables.
setup.py: increase minimum python version to 3.6.

### CODE REVIEW

File: docs/design.rst [hunk 1]
```diff
@@ -26,12 +30,35 @@ In addition interactions with the instance are covered by a standard set of comm
...
+Exceptions
+==========
 
 All exceptions from underlying libraries are passed directly through for the end-user. There are a large number of exceptions to catch and possibilities, not to mention that they can change over time. By not catching them it informs the user that issues are found with what they are doing instead of hiding it from them.
 
-## Logging
+Logging
+=======
 
 Logging is setup using the standard logging module. It is up to the user to setup their logging configuration and set the appropriate level.
```
Comment (s-makin) [inline docs/design.rst:41]:
```suggestion
Logging is set up using the standard logging module. It is up to the user to set up their logging configuration and set the appropriate level.
```

File: docs/design.rst [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+Images
+======
 
 Instances are expected to use the latest daily image, unless another image is specifically requested.
 
-### cloud-init
+cloud-init
+----------
 
 The images are expected to have cloud-init in them to properly start. When an instance is started or during launch, the instance is checked for the boot complete file that cloud-init produces.
```
Comment (s-makin) [inline docs/design.rst:14]:
```suggestion
The images are expected to have cloud-init in them to properly start. When an instance is started, or during launch, the instance is checked for the boot complete file that cloud-init produces.
```

**Review by s-makin — APPROVED** [review]

File: docs/design.rst (hunk 2 above)
Comment (s-makin) [reply docs/design.rst:14]: Just a couple of minor nits from the original document. Otherwise LGTM!

**Review by s-makin — APPROVED** [review]

**Review by orndorffgrant — APPROVED** [review]

**Review by blackboxsw — APPROVED** [review]
LGTM.

**Pull request merged**